x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/TemplateUtils.java

```diff
@@ -57,11 +58,18 @@ public static void loadTemplateIntoMap(String resource, Map<String, IndexTemplat
      * Loads a built-in template and returns its source.
      */
     public static String loadTemplate(String resource, String version, String versionProperty) {
+        return loadTemplate(resource, version, versionProperty, Collections.emptyMap());
+    }
+
+    /**
+     * Loads a built-in template and returns its source after replacing given variables.
+     */
+    public static String loadTemplate(String resource, String version, String versionProperty, Map<String, String> variables) {
         try {
             BytesReference source = load(resource);
+            source = replaceVariables(source, version, versionProperty, variables);
```

Comment:
@gwbrown Here are the changes I made to allow variable replacement on template loading. Could you take a look please when you have some time?